### PR TITLE
feat: support update priorityClassName

### PR DESCRIPTION
**Pull request by merryzhou**
Signed-off-by: merryzhou <merryzhou0882@gmail.com>

### CODE REVIEW

**Comment by merryzhou** [comment]
https://github.com/volcano-sh/volcano/issues/1097

File: pkg/controllers/job/job_controller_actions.go [hunk 1]
```diff
@@ -507,19 +508,29 @@ func (cc *jobcontroller) createOrUpdatePodGroup(job *batch.Job) error {
 		return nil
 	}
 
-	if pg.Spec.MinMember != job.Spec.MinAvailable {
+	pgShouldUpdate := false
```
Comment (Thor-wl) [inline pkg/controllers/job/job_controller_actions.go:511]: Have you tested the effect of priorityClassName update on actions such as `preempt` ?

Comment (merryzhou) [reply pkg/controllers/job/job_controller_actions.go:511]: I have tested allocate action， it‘s ok。 And all actions use priority(/priorityClassName) in similar ways, so theoretically there is no problem.

**Comment by Thor-wl** [comment]
Please resovle the conflictions

**Pull request merged**